dashboard/lib/pd/mimeo_rest_client.rb

```diff
@@ -142,14 +151,32 @@ def self.final_status?(status)
 
   private
 
+  # Determines if the error is a known retriable error,
+  # a spurious internal server error in the Mimeo API
+  # @param [RestClient::ExceptionWithResponse] error
+  # @return [Boolean] true if the error is retryable, otherwise false
+  def retryable_error?(error)
+    error.try(:response).try(:body).try(:include?, RETRYABLE_ERROR_TEXT)
+  end
+
+  # Executes the block with retries for retriable errors
+  def with_retries
+    exception_cb = ->(exception) {raise exception unless retryable_error?(exception)}
+    Retryable.retryable(tries: @max_attempts, exception_cb: exception_cb) do
+      yield
```

Comment:
Worth logging # of retries as something to measure? Excessive retries is something we should be bringing up to Mimeo